cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -127,6 +126,33 @@ public void deleteRepository(FullHttpRequest request, HttpResponder responder,
                                                               namespaceId));
   }
 
+  /**
+   * Lists applications found in linked repository for the given namespace.
+   * Returns the application name and git-filehash for each application config file found
+   * <pre>
+   * {@code
+   * {
+   *  apps:[
+   *    {
+   *      "name": "xyz",
+   *      "fileHash": "abc"
+   *    }
+   *  ]
+   * }
+   * }
+   *
+   * </pre>
+   */
+  @GET
+  @Path("/apps")
+  public void listApplications(FullHttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId) throws Exception {
+    checkSourceControlFeatureFlag();
```

Comment:
nit: Should we just skip installing the service if it's disabled?

Reply:
Out of scope for the PR. But will check in a later PR